packages/yourloops/components/patient-list/patient-list.hook.tsx

```diff
@@ -0,0 +1,293 @@
+  onRowClick: (params: GridRowParams) => void
+  toggleColumnVisibility: (columnName: PatientListColumns) => void
+}
+
+export const usePatientListHook = (): PatientListHookReturns => {
+  const { t } = useTranslation()
+  const { classes } = usePatientListStyles()
+  const { getFlagPatients, user } = useAuth()
+  const { getPatientById, filterPatients } = usePatientContext()
+  const { getUserName } = useUserName()
+  const { sortByUserName } = useSortComparatorsHook()
+  const navigate = useNavigate()
+  const gridApiRef = useGridApiRef()
+  const queryParams = useQueryParams()
+  const trNA = t('N/A')
+
+  const [selectedTab, setSelectedTab] = useState<PatientListTabs>(PatientListTabs.Current)
+  const [selectedFilter, setSelectedFilter] = useState<PatientListFilters>(PatientListFilters.All)
+  const [inputSearch, setInputSearch] = useState<string>('')
+  const [patientToRemoveForHcp, setPatientToRemoveForHcp] = useState<Patient | null>(null)
+  const [patientToRemoveForCaregiver, setPatientToRemoveForCaregiver] = useState<UserToRemove | null>(null)
+  const [paginationModel, setPaginationModel] = useState<GridPaginationModel>({ pageSize: 10, page: 0 })
+  const [columnsVisibility, setColumnsVisibility] = useState<GridColumnVisibilityModel>({
+    [PatientListColumns.Flag]: true,
+    [PatientListColumns.System]: true,
+    [PatientListColumns.Patient]: true,
+    [PatientListColumns.TimeOutOfRange]: true,
+    [PatientListColumns.SevereHypoglycemia]: true,
+    [PatientListColumns.DataNotTransferred]: true,
+    [PatientListColumns.LastDataUpdate]: true,
+    [PatientListColumns.Messages]: user.isUserHcp() ?? false,
+    [PatientListColumns.Actions]: true
+  })
+
+  const flaggedPatients = getFlagPatients()
+
+  const filteredPatients = useMemo(() => {
+    const filteredPatients = filterPatients(selectedFilter, inputSearch, flaggedPatients)
+    return PatientUtils.computeFlaggedPatients(filteredPatients, flaggedPatients)
+  }, [selectedFilter, filterPatients, flaggedPatients, inputSearch])
+
+  const onChangingTab = (newTab: PatientListTabs): void => {
+    setSelectedTab(newTab)
+    switch (newTab) {
+      case PatientListTabs.Current:
+        setSelectedFilter(PatientListFilters.All)
+        return
+      case PatientListTabs.Pending:
+        setSelectedFilter(PatientListFilters.Pending)
+    }
+  }
+
+  const onClickRemovePatient = useCallback((patientId: string): void => {
+    const patient = getPatientById(patientId)
+    if (user.isUserHcp()) {
+      setPatientToRemoveForHcp(patient)
+      return
+    }
+    setPatientToRemoveForCaregiver({
+      id: patient.userid,
+      fullName: getPatientFullName(patient),
+      email: patient.profile.email
+    })
+  }, [getPatientById, user])
+
+  const onCloseRemoveDialog = (): void => {
+    setPatientToRemoveForHcp(null)
+    setPatientToRemoveForCaregiver(null)
+  }
+
+  const onRowClick = (params: GridRowParams): void => {
+    navigate(`/patient/${params.id}`)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** This might avoid a render in the router ‏
```suggestion
    navigate(`/patient/${params.id}/dashboard`)
```